srcpkgs/mailnag/template

```diff
@@ -0,0 +1,21 @@
+# Template file for 'mailnag'
+pkgname=mailnag
+version=2.2.0
+revision=1
+build_style=python3-module
+hostmakedepends="python3 gettext"
+# a libsecret client (GNOME Keyring, Kwallet...) is needed
+# a libnotify client is needed (if you want to use the libnotify extention)
+# You should also have a working syslog (https://docs.voidlinux.org/config/services/logging.html) if you do not want error messages about syslog
+depends="python3 python3-gobject python3-dbus python3-xdg libsecret gtk+3 $(vopt_if sound gst-plugins-base1) $(vopt_if notify libnotify)"
```

Comment:
Listing python3 is not needed when some module is listed.

Reply:
You mean the build style? If yes, I have thought that I need to specify this, because the [manual](https://github.com/Cnoob41/void-packages/blob/master/CONTRIBUTING.md) says that:

> Please note that required packages to execute a `build_style` script must be defined via `$hostmakedepends`.

Reply:
In `depends` a module that mailnag depend on will depend on interpreter, so no need to add it explicitly.
For hostmakedepends you're right, please leave as is.

Reply:
What do you mean by that?

Reply:
Replace `depends="python3 python3-gobject`... with `depends="python3-gobject`...

Reply:
But the program depends on `python3`. It will not run without it.

Reply:
So does python3-xdg.

mailnag depends on python3-xdg.
python3-xdg depends on python3.
Therefore, python3 will be installed when mailnag is installed.

Reply:
Oh. But this will not have any effect on this, am I right? I agree with you and I have removed this dependency, but I'd like to know if it will matter if I would add a package `x` and `y` to dependencies if I wouldn't notice that package `x` depends on package `y`.

Reply:
Generally all dependencies should be added as stated by upstream, to be prepared to dropping dependencies by other packages. Python interpreter is one of few cases where it's obvious that it is needed by modules, so listing it is noise.

Dependencies detected and listed in build log by 04-generate-runtime-deps hook aren't listed either. There are none for this package as it is written in python, but for compiled packages it's often all dependencies.

Reply:
Ok.